sdk/trace/span.go

```diff
@@ -265,17 +276,44 @@ func (s *recordingSpan) addOverCapAttrs(attrs []attribute.KeyValue) {
 			continue
 		}
 
-		if len(s.attributes) >= s.tracer.provider.spanLimits.AttributeCountLimit {
+		if len(s.attributes) >= limit {
 			// Do not just drop all of the remaining attributes, make sure
 			// updates are checked and performed.
 			s.droppedAttributes++
```

Comment:
I know this isn't part of your change, but is this comment still accurate?

Reply:
Yes. We are still checking for any duplicate attrs and updating accordingly. Incrementing the `droppedAttributes` count by `len(s.attributes)-limit`, setting `s.attributes = s.attributes[:limit]`, and then returning would be the alternative (an incorrect thing to do).